### PR TITLE
Catch ObjectDisposedException in IconTintColorBehavior

### LINKED ISSUE #1819: [BUG] IconTintColorBehavior crash on android.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

IconTintColorBehavior crash while changing tint color.

System.ObjectDisposedException: ObjectDisposed_Generic
ObjectDisposed_ObjectName_Name, AndroidX.AppCompat.Widget.AppCompatImageView
   at Java.Interop.JniPeerMembers.AssertSelf(IJavaPeerable )
   at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualVoidMethod(String , IJavaPeerable , JniArgumentValue* )
   at Android.Widget.ImageView.SetColorFilter(ColorFilter )
   at CommunityToolkit.Maui.Behaviors.IconTintColorBehavior.<ApplyTintColor>g__SetImageViewTintColor|2_0(ImageView image, Color color)

### Expected Behavior

IconTintColorBehavior does not crash. 

### Steps To Reproduce

Unfortunately, I have some nuget problems with adding package, but you should be able to reproduce it by changing tint color.

### Link to public reproduction project repository

no link currently

### Environment

```markdown
- .NET MAUI CommunityToolkit:
- OS:
- .NET MAUI:
```

### Anything else?

I found fix for this. Just remove PropertyChanged event handler in OnDetachedFrom method. In previous version handler is added, but not removed.

public partial class WorkaroundTintedImageBehavior
{
    private View? _bindable;
    private AView? _platformView;
    
    protected override void OnAttachedTo(View bindable, AView platformView)
    {
        _bindable = bindable;
        _platformView = platformView;
        
        bindable.PropertyChanged += OnElementPropertyChanged;
        ApplyTintColor(platformView);
        PropertyChanged += OnTintedImagePropertyChanged;
    }

    protected override void OnDetachedFrom(View bindable, AView platformView)
    {
        ClearTintColor(bindable, platformView);
        bindable.PropertyChanged -= OnElementPropertyChanged;
        PropertyChanged -= OnTintedImagePropertyChanged;
    }

    private void OnTintedImagePropertyChanged(object? sender, PropertyChangedEventArgs e)
    {
        if (e.PropertyName == TintColorProperty.PropertyName && _bindable is not null && _platformView is not null)
        {
            ApplyTintColor(_platformView);
        }
    }

    private void ApplyTintColor(AView control)
    {
        var color = TintColor;

        switch (control)
        {
            case ImageView image:
                SetImageViewTintColor(image, color);
                break;
            case AButton button:
                SetButtonTintColor(button, color);
                break;
            default:
                throw new NotSupportedException($"{nameof(WorkaroundTintedImageBehavior)} only currently supports Android.Widget.Button and {nameof(ImageView)}.");
        }

        static void SetImageViewTintColor(ImageView image, Color? color)
        {
            if (color is null)
            {
                image.ClearColorFilter();
                return;
            }

            image.SetColorFilter(new PorterDuffColorFilter(color.ToPlatform(), PorterDuff.Mode.SrcIn ?? throw new InvalidOperationException("PorterDuff.Mode.SrcIn should not be null at runtime.")));
        }

        static void SetButtonTintColor(AButton button, Color? color)
        {
            var drawables = button.GetCompoundDrawables().Where(d => d is not null);

            if (color is null)
            {
                foreach (var img in drawables)
                {
                    img.ClearColorFilter();
                }

                color = Colors.Transparent;
            }

            foreach (var img in drawables)
            {
                img.SetTint(color.ToPlatform());
            }
        }
    }

    private void OnElementPropertyChanged(object? sender, PropertyChangedEventArgs args)
    {
        if (args.PropertyName is not string propertyName
            || sender is not View bindable
            || bindable.Handler?.PlatformView is not AView platformView)
        {
            return;
        }

        if (!propertyName.Equals(Image.SourceProperty.PropertyName, StringComparison.Ordinal)
            && !propertyName.Equals(ImageButton.SourceProperty.PropertyName, StringComparison.Ordinal))
        {
            return;
        }

        ApplyTintColor(platformView);
    }

    private void ClearTintColor(View element, AView control)
    {
        element.PropertyChanged -= OnElementPropertyChanged;
        switch (control)
        {
            case ImageView image:
                image.ClearColorFilter();
                break;
            case AButton button:
                foreach (var drawable in button.GetCompoundDrawables())
                {
                    drawable?.ClearColorFilter();
                }

                break;
        }
    }
}

### PR DESCRIPTION
**Pull request by Kay-STL**
### Description of Change ###
 
Related to the last PR for this issue (#2012). This PR explicitly catches ObjectDisposedException in the same way as [TouchBehavior](https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/TouchBehavior.android.cs#L105). Not an optimal solution, but it's a simple fix and makes absolutely sure that it won't crash the app.

 ### Linked Issues ###
 

 - Fixes #1819 

 ### PR Checklist ###
 
 - [X] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [X] Rebased on top of `main` at time of PR
 - [X] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

Android only, occurs in a production application. After the previous PR the crash couldn't be reproduced, but a few occurrences were spotted in error reporting.

### CODE REVIEW

File: src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.android.cs [hunk 1]
```diff
@@ -109,22 +117,29 @@ static void ClearTintColor(AndroidView? nativeView)
...
+					{
+						drawable.ClearColorFilter();
+					}
+
+					break;
+			}
+		}
+		catch (ObjectDisposedException)
+		{
+			Trace.WriteLine("IconTintColorBehavior is already disposed.");
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.android.cs:142]:
```suggestion
```

File: src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.android.cs [hunk 2]
```diff
@@ -44,22 +45,29 @@ static void ApplyTintColor(AndroidView? nativeView, Color? tintColor)
...
 
-			default:
-				throw new NotSupportedException($"{nameof(IconTintColorBehavior)} only currently supports Android.Widget.Button and {nameof(ImageView)}.");
+				default:
+					throw new NotSupportedException($"{nameof(IconTintColorBehavior)} only currently supports Android.Widget.Button and {nameof(ImageView)}.");
+			}
+		}
+		catch (ObjectDisposedException)
+		{
+			Trace.WriteLine("IconTintColorBehavior is already disposed.");
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.android.cs:70]:
```suggestion
```

**Review by brminnick — COMMENTED** [review]
Thanks! Let's just remove these comments so we don't accidentally confuse devs in their traces.

**Review by brminnick — APPROVED** [review]

**Comment by Kay-STL** [comment]
Looks like the pipeline is broken - fails every time with:
```
ILLINK : warning MT0079: The recommended Xcode version for Microsoft.MacCatalyst 17.5.8018 is Xcode 15.4 or later. The current Xcode version (found in /Applications/Xcode_15.2.0.app/Contents/Developer) is 15.2. [/Users/runner/work/1/s/samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj::TargetFramework=net8.0-maccatalyst]
ILLink : unknown error IL7000: An error occured while executing the custom linker steps. Please review the build log for more information. [/Users/runner/work/1/s/samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj::TargetFramework=net8.0-ios]
ILLINK : error MT0180: This version of Microsoft.iOS requires the iOS 17.5 SDK (shipped with Xcode 15.4). Either upgrade Xcode to get the required header files or set the managed linker behaviour to Link Framework SDKs Only in your project's iOS Build Options > Linker Behavior (to try to avoid the new APIs). [/Users/runner/work/1/s/samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj::TargetFramework=net8.0-ios]
ILLINK : error MT2301: The linker step 'Setup' failed during processing: This version of Microsoft.iOS requires the iOS 17.5 SDK (shipped with Xcode 15.4). Either upgrade Xcode to get the required header files or set the managed linker behaviour to Link Framework SDKs Only in your project's iOS Build Options > Linker Behavior (to try to avoid the new APIs). [/Users/runner/work/1/s/samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj::TargetFramework=net8.0-ios]
/Users/runner/.nuget/packages/microsoft.net.illink.tasks/8.0.7/build/Microsoft.NET.ILLink.targets(87,5): error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false. [/Users/runner/work/1/s/samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj::TargetFramework=net8.0-ios]
```

**Comment by Kay-STL** [comment]
Should auto-merge once #2100 is merged and this branch is synced.

**Pull request merged**